src/blockchain/handler/swap.ts

```diff
@@ -21,58 +22,65 @@ export default class SwapHandler {
 
     private contracts: any;
     private adapters: any;
+    private priceService: PriceService;
 
     constructor() {
         this.swapService = new SwapService();
         this.emailService = new EmailService();
         this.exchange = new Exchange();
         this.contracts = getContracts();
         this.adapters = getAdapters();
+        this.priceService = new PriceService();
     }
 
     async onSwap(inputSwap, maxTries = RETRY_COUNT) {
-        logInfo(`SWAP_RECEIVED`, inputSwap);
+        try {    
+            logInfo(`SWAP_RECEIVED`, inputSwap);
 
-        const isProcessed = await this.swapService.findByIdAndNetwork(inputSwap.id, inputSwap.network);
+            const isProcessed = await this.swapService.findByIdAndNetwork(inputSwap.id, inputSwap.network);
 
-        if (!isProcessed) {
-            const adapter = this.adapters[inputSwap.outputNetwork];
-            const contract = this.contracts[inputSwap.outputNetwork];
+            if (!isProcessed) {
+                const adapter = this.adapters[inputSwap.outputNetwork];
+                const contract = this.contracts[inputSwap.outputNetwork];
 
-            const outputSwap = adapter.createSwapFromInput(inputSwap);
+                let outputSwap = adapter.createSwapFromInput(inputSwap);
+                outputSwap = this.priceService.adjustPrice(outputSwap);
```

Comment:
This could become `outputSwap.inputAmount = this.getCurrentInputPrice(outputSwap);`